Zed pulp & package repo config

Alex-Welsh opened the pull request:
This PR mostly contains config changes for Pulp and Zed-specific package repos (again, cleaving out a bunch of CS8/RL8/focal stuff).

There are also some commits for kolla container image building, since many of the changes were made at the same time. Zed container image builds still require some more changes after these, but this PR is already plenty big enough.

Alex-Welsh committed:
Update repos for Zed

Alex-Welsh committed:
Update pulp.yml for Zed release

Removed references to distributions/releases that are no longer
supported (CentOS 8 Stream, Rocky Linux 8, and Ubuntu Focal 20.04).

Added OpenSearch and OpenSearch Dashboards for Rocky Linux 9

Alex-Welsh committed:
Fix kolla container image builds

Alex-Welsh committed:
Update kolla config for container image builds

Alex-Welsh committed:
Update apt.yml for Zed release

Removed references to focal and enable jammy repos

Alex-Welsh committed:
Update dnf.yml for Zed release

Removed references to CentOS 8 Stream and Rocky Linux 8

Alex-Welsh committed:
Remove docker package config from AIO environment

Configuration for the docker repo is managed in dnf.yml and does not
need to be explicitly included in the AIO environment.

Alex-Welsh commented:
I can try to subdivide this further if it would help

markgoddard commented:
> I can try to subdivide this further if it would help

I can handle it :D

markgoddard commented on etc/kayobe/apt.yml:
```diff
@@ -63,12 +63,12 @@ stackhpc_apt_repositories:
 # when replacing the distribution repositories via apt_repositories.
 # Default is false.
-apt_disable_sources_list: "{{ ansible_facts['distribution_release'] != 'jammy' }}"
+apt_disable_sources_list: "true"
```
```suggestion
apt_disable_sources_list: true
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/environments/ci-aio/kolla/globals.yml:
```diff
@@ -9,7 +9,7 @@ nova_compute_virt_type: qemu
 
-docker_yum_baseurl: "{{ stackhpc_repo_docker_url }}"
+docker_yum_baseurl: "{{ stackhpc_repo_centos_stream_9_docker_url }}"
 docker_yum_gpgkey: "https://download.docker.com/linux/centos/gpg"
```
I don't think we need these since docker is in dnf.yml

Alex-Welsh replied:
Removed

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -4,9 +4,6 @@
 # Kolla installation.
 
-# Type of Kolla control installation. One of 'binary' or 'source'.
-#kolla_ctl_install_type:
```
This refers to the installation of the kolla python package, not the packages in containers.

Alex-Welsh replied:
Odd this has cropped up twice. Reverted

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -66,8 +63,8 @@ kolla_ansible_source_version: "{{ stackhpc_kolla_ansible_source_version }}"
-# Kolla base container image distribution. Options are "centos", "debian",
-# "rocky", "ubuntu". Default is {{ os_distribution }}.
+# Kolla base container image distribution. Options are "rocky", "ubuntu".
+# Default is {{ os_distribution }}.
```
Probably best to stick with upstream for these or we'll end up in merge hell.

Alex-Welsh replied:
Good point. Reverted

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -336,16 +272,6 @@ kolla_build_blocks:
         rm -f liblasso3_2.7.0-2build3_amd64.deb
     {% endif %}
     {% endraw %}
-  nova_base_footer: |
```
@cityofships do you know/remember whether the liblasso fix is still necessary on Jammy?

Alex-Welsh replied:
I don't have the cherry pick here but I did actually have to remove this in the zed-init branch to get keystone to build https://github.com/stackhpc/stackhpc-kayobe-config/commit/d84a6214505d70c4eb3bae5d2a3bda339fcade44

Alex-Welsh replied:
I was going to add it to a final _kolla image build fixes_ PR but can put it in here if you want

markgoddard replied:
That's fine, as long as it goes in eventually.

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -360,7 +286,7 @@ kolla_build_customizations_common:
     - /additions/*
 
-kolla_build_customizations_centos:
+kolla_build_customizations_rocky: # TODO: Check whether this is actually correct. It seems that it was previously being skipped for rocky
```
@mnasiadka are the OVN/OVS bits still relevant on RL9?

mnasiadka replied:
If we won't pin to OVS 2.17 and OVN whatever is latest, we might get OVS 3.1 - which is not LTS, but still should work (and give us some new features). I think it should be some joint decision if we are going OVS 3.1 path or not.
@bbezak @oneswig any insights?

bbezak replied:
I'm happy to use ovs 3.1 - however we need to test that before deploying to prod branches

markgoddard replied:
We're not applying these build customisations to RL9 on Yoga. Does that mean we may get OVS 3.1 there too?

mnasiadka replied:
Well, for now it's OVS 2.17 and OVN 22.09 - but that's due to pinning in virtual rdo-openvswitch and rdo-ovn packages, but RDO is in control now - not us ;-)

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -226,232 +208,7 @@ stackhpc_pulp_repo_rocky_9_highavailability_version: "{{ lookup('vars', 'stackhp
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-
-  - name: Grafana
```
Looks like we've lost Grafana.

Alex-Welsh replied:
Ah, yes. I think this is one of those things I missed then fixed in a later commit. I'll correct it here

Alex-Welsh replied:
Fixed

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
     sync_policy: mirror_complete
     state: present
     required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: OpenSearch for RHEL 9
```
```suggestion
  - name: OpenSearch 2.x
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
     state: present
     required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: OpenSearch for RHEL 9
+    url: "{{ stackhpc_release_pulp_content_url }}/pulp/content/opensearch/2.x/yum/{{ stackhpc_pulp_repo_opensearch_2_x_version }}"
```
```suggestion
    url: "{{ stackhpc_release_pulp_content_url }}/opensearch/2.x/yum/{{ stackhpc_pulp_repo_opensearch_2_x_version }}"
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
+    sync_policy: mirror_complete
+    state: present
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: OpenSearch Dashboards for RHEL 9
```
```suggestion
  - name: OpenSearch Dashboards 2.x
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -621,6 +366,24 @@ stackhpc_pulp_repository_rpm_repos:
+    state: present
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: OpenSearch Dashboards for RHEL 9
+    url: "{{ stackhpc_release_pulp_content_url }}/pulp/content/opensearch-dashboards/2.x/yum/{{ stackhpc_pulp_repo_opensearch_dashboards_2_x_version }}"
```
```suggestion
    url: "{{ stackhpc_release_pulp_content_url }}/opensearch-dashboards/2.x/yum/{{ stackhpc_pulp_repo_opensearch_dashboards_2_x_version }}"
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -631,125 +394,17 @@ stackhpc_pulp_publication_rpm_development: "{{ stackhpc_pulp_distribution_rpm_de
-    repository: ELK repository for 7.x packages
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "grafana-development"
```
ditto

Alex-Welsh replied:
Interesting that this wasn't being synced for RL9 to begin with. Is it necessary?

Alex-Welsh replied:
Fixed

markgoddard replied:
https://github.com/stackhpc/stackhpc-kayobe-config/pull/605/files

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -826,129 +481,31 @@ stackhpc_pulp_distribution_rpm_development:
-    distribution: "extra-packages-for-enterprise-linux-8-x86_64-development"
+  - name: "opensearch-2-x-development"
+    base_path: "opensearch/2.x/yum/development"
+    repository: OpenSearch for RHEL 9
```
```suggestion
    repository: OpenSearch 2.x

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -826,129 +481,31 @@ stackhpc_pulp_distribution_rpm_development:
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: "opensearch-dashboards-2-x-development"
+    base_path: "opensearch-dashboards/2.x/yum/development"
+    repository: OpenSearch for RHEL 9
```
```suggestion
    repository: OpenSearch Dashboards 2.x

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -826,129 +481,31 @@ stackhpc_pulp_distribution_rpm_development:
-    repository: ELK repository for 7.x packages
-    state: present
-    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
-  - name: "grafana-production"
```
ditto

Alex-Welsh replied:
Fixed

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -1025,6 +582,16 @@ stackhpc_pulp_distribution_rpm_production:
     required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: "opensearch-2-x-production"
+    base_path: "opensearch/2.x/yum/production"
+    repository: OpenSearch for RHEL 9
```
```suggestion
    repository: OpenSearch 2.x
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -1025,6 +582,16 @@ stackhpc_pulp_distribution_rpm_production:
+    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_9 | bool }}"
+  - name: "opensearch-dashboards-2-x-production"
+    base_path: "opensearch-dashboards/2.x/yum/production"
+    repository: OpenSearch for RHEL 9
```
```suggestion
    repository: OpenSearch Dashboards 2.x
```

Alex-Welsh replied:
Done

Alex-Welsh committed:
Fix misc edits from code review

Explanation of these changes can be found here:
https://github.com/stackhpc/stackhpc-kayobe-config/pull/600

markgoddard commented:
Has some merge conflicts

Alex-Welsh commented:
> Has some merge conflicts

Rebased & resolved

markgoddard reviewed (commented):
LGTM pending open questions for @cityofships and @mnasiadka
